katran/lib/KatranLb.h

```diff
@@ -203,6 +203,15 @@ class KatranLb {
    */
   uint32_t getVipFlags(const VipKey& vip);
 
+  /**
+   * @param std::string& real to get flags from
+   * @return uint8_t flags of this real
+   *
+   * helper function to return flags of specified real
+   * could throw if specified real doesn't exist
+   */
+  uint8_t getRealFlags(const std::string& real);
```

Comment:
i think this is non needed. we already have getRealsForVip which returns NewReal where you have added flag field.

returning flag for real does not gave user enough information as he does not know about vip's flag. in comparision getRealsForVip both contains vip's flags (as vip is being used in that function as argument) + returns real's flag. this is enough information for end user to figure out if vip + real has local optimization enabled